src/R/Components/Impl/InteractiveWorkflow/Implementation/RSessionCallback.cs

```diff
@@ -134,10 +134,14 @@ public Task<PlotDeviceProperties> PlotDeviceCreate(Guid deviceId, CancellationTo
             }
 
             try {
-                cancellationToken.ThrowIfCancellationRequested();
-                _fileSystem.FileWriteAllBytes(localPath, data);
+                var message = Resources.Progress_FetchingFile.FormatInvariant(remoteFileName);
+                _coreShell.ProgressDialog.Show(async (progress, ct) => {
+                    using (DataTransferSession dts = new DataTransferSession(_session, _fileSystem)) {
+                        await dts.FetchAndDecompressFileAsync(remoteBlobId, localPath, progress, message, cancellationToken);
+                    }
+                }, message);
             } catch (Exception ex) {
-                await _coreShell.ShowErrorMessageAsync(Resources.Error_UnableSaveFile.FormatInvariant(localPath, ex.Message), cancellationToken);
+                await _coreShell.ShowErrorMessageAsync(Resources.Error_UnableToTransferFile.FormatInvariant(localPath, ex.Message), cancellationToken);
```

Comment:
Since we're on the UI thread already, `_coreShell.ShowErrorMessage` should be enough